apache/ambari
# [AMBARI-22904] Revised mpack APIs

## What changes were proposed in this pull request?
Revised mpack APIs as per the new layout.
1. POST /mpacks -> download mpack.json and tar.gz individually in a single call.
2. GET /mpacks -> Read from in-memory map instead of a database hit.
3. Refactor packlets -> Modules with additional information in line with new mpack.json

## How was this patch tested?
1. Register a new Mpack.
2. Fetch the registered mpacks.
3. Restart ambari-server and check if mpacks are persisted during next get call.
4. Test stack apis for regression.
5. Verify all steps upto cluster deployment goes through.
6. Run build on project to ensure no compilation issues.

Please review [Ambari Contributing Guide](https://cwiki.apache.org/confluence/display/AMBARI/How+to+Contribute) before opening a pull request.
